src/hb-subset-plan.cc

```diff
@@ -853,6 +854,86 @@ _normalize_axes_location (hb_face_t *face, hb_subset_plan_t *plan)
   }
   plan->all_axes_pinned = !axis_not_pinned;
 }
+
+static void
+_update_instance_metrics_map_from_cff2 (hb_subset_plan_t *plan)
+{
+  if (!plan->normalized_coords) return;
+  OT::cff2::accelerator_t cff2 (plan->source);
+  if (!cff2.is_valid ()) return;
+
+  hb_font_t *font = nullptr;
+  if (unlikely (!plan->check_success (font = _get_hb_font_with_variations (plan))))
```

Comment:
add a hb_font_destroy (font) before returning, since check_success can return false even if font creation failed in the event the plan was already in error. See: https://github.com/harfbuzz/harfbuzz/pull/4187/files